Framework/Kernel/src/VectorHelper.cpp

```diff
@@ -27,6 +27,10 @@ int DLLExport createAxisFromRebinParams(const std::vector<double> &params,
                                         std::vector<double> &xnew,
                                         const bool resize_xnew,
                                         const bool full_bins_only) {
+  if (params.size() == 1) {
```

Comment:
I am not sure I understand what you did here. Don't other algorithms like `Rebin` support such parameters?

Reply:
They do but in case of `Rebin`, the min and max X are known. The `SofQW*` algorithms convert to momentum transfer and it's a bit tedious to guess suitable Qmin and Qmax beforehand.